module_utils/storage_lsr/argument_validator.py

```diff
@@ -0,0 +1,171 @@
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+class ArgValidator(object):
+
+    @classmethod
+    def _validate_list(cls, value):
+        if isinstance(value, list):
+            return value
+
+        if isinstance(value, str):
+            return value.split(",")
+
+        if isinstance(value, int) or isinstance(value, float):
+            return [value]
+
+        raise TypeError("'%s' has to be a list" % value)
+
+    @classmethod
+    def _validate_str(cls, value):
+        return str(value)
```

Comment:
see other comment about strings

Reply:
In this check, no actual validation is done, because anything can be converted to string. This will just make sure that the no matter what input was the result is string type.
This should be doable by standard `str()`.